### ISSUE TITLE
🐛 Bug Report: cannot build functions

### ISSUE THREAD

**Issue by superseby2** [appwrite/appwrite#2989]
### 👟 Reproduction steps

using cli

appwrite deploy function
ℹ Info Deploying function <functionID> ( 6239f0584685eb7dec52 )
✓ Success Deployed <functionID> ( 6239f0584685eb7dec52 )

### 👍 Expected behavior

- The function is successfully deployed

### 👎 Actual Behavior

 - function not deployed
 - error is   
 - 
`Docker Error: WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.`

Eventhough it is a Warning the function is not run

### 🎲 Appwrite version

Version 0.13.x

### 💻 Operating system

Linux

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by stnguyen90** [comment]
Related Issue: https://github.com/utopia-php/orchestration/pull/16

Workaround: https://medium.com/geekculture/appwrite-frequently-asked-questions-374ce81513fe#00aa

**Comment by stnguyen90** [comment]
Unfortunately, I don't think the fix made it into 0.13.4: https://github.com/appwrite/appwrite/blob/0.13.4/composer.lock#L2468

**Comment by ChristofFritz** [comment]
Thank you @stnguyen90 you saved my evening :D

**Comment by superseby2** [comment]
Thx @stnguyen90  Sadly I could not access your read the article (I don't wanna be forced to login to this platform)
In the meantime I followed https://fabianlee.org/2020/01/18/docker-placing-limits-on-container-memory-using-cgroups/ and enabled limits on containers via a grub settings. 

IMHO if the orchestrator is setting limits when running containers, as a user, I would prefer to be notified that this security measure cannot be applied and have a way to disable it rather than not knowing at all.
Hence, in that specific case, the function not being able to run because of this specific warning is fine as long as I can disable memory control via a vairable.

**Comment by stnguyen90** [comment]
> Thx @stnguyen90 Sadly I could read the article (I don't wanna be forced to login to this platform) In the meantime I followed https://fabianlee.org/2020/01/18/docker-placing-limits-on-container-memory-using-cgroups/ and enabled limits on containers via a grub settings.
> 
> IMHO if the orchestrator is setting limits when running containers, as a user, I would prefer to be notified that this security measure cannot be applied and have a way to disable it rather than not knowing at all. Hence, in that specific case, the function not being able to run because of this specific warning is fine as long as I can disable memory control via a vairable (or adjust the host setup as I did)

You might be able to get to the article if you open in incognito 😉. There's also a workaround in the article that involves changing the function memory and swap environment variables to 0.

**Comment by superseby2** [comment]
Thx Steven!

Quoting Steven (2022-03-24 15:08:09)
>     Thx @stnguyen90 Sadly I could read the article (I don't wanna be forced to
>     login to this platform) In the meantime I followed https://fabianlee.org/
>     2020/01/18/docker-placing-limits-on-container-memory-using-cgroups/ and
>     enabled limits on containers via a grub settings.
> 
>     IMHO if the orchestrator is setting limits when running containers, as a
>     user, I would prefer to be notified that this security measure cannot be
>     applied and have a way to disable it rather than not knowing at all. Hence,
>     in that specific case, the function not being able to run because of this
>     specific warning is fine as long as I can disable memory control via a
>     vairable (or adjust the host setup as I did)
> 
> You might be able to get to the article if you open in incognito 😉. There's
> also a workaround in the article that involves changing the function memory and
> swap environment variables to 0.
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you authored the thread.*Message ID: <appwrite/
> ***@***.***>
> 

-- 
Sebastien Pastor
CEO AfreelancersWorld.com

Phone: +33 662 183 484
LinkedIn: www.linkedin.com/in/sebastienpastor
AFW: tech.afreelancersworld.com/u/elsuperseby

[email propulsed by notmuch/afew/alot]

**Comment by acalatrava** [comment]
I tried the second workaround (setting the env vars to 0) but I receive another error:
```
Docker Error: docker: Error response from daemon: You should always set the Memory limit when using Memoryswap limit, see usage.
See 'docker run --help'.
```

I don't want to reboot my server since I'm using it for other stuff (last reboot was 445 days ago) so my only workaround was to fork Appwrite and comment these lines:
https://github.com/acalatrava/appwrite/commit/f8534a1af026cffb784c7b4564810bafcb6032c9

**Comment by OHeroJ** [comment]
> ### 👟 Reproduction steps
> using cli
> 
> appwrite deploy function ℹ Info Deploying function ( 6239f0584685eb7dec52 ) ✓ Success Deployed ( 6239f0584685eb7dec52 )
> 
> ### 👍 Expected behavior
> * The function is successfully deployed
> 
> ### 👎 Actual Behavior
> * function not deployed
> * error is
> 
> `Docker Error: WARNING: Your kernel does not support swap limit capabilities or the cgroup is not mounted. Memory limited without swap.`
> 
> Eventhough it is a Warning the function is not run
> 
> ### 🎲 Appwrite version
> Version 0.13.x
> 
> ### 💻 Operating system
> Linux
> 
> ### 🧱 Your Environment
> _No response_
> 
> ### 👀 Have you spent some time to check if this issue has been raised before?
> * [x]  I checked and didn't find similar issue
> 
> ### 🏢 Have you read the Code of Conduct?
> * [x]  I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

+1, same error

**Comment by christyjacob4** [comment]
These issues were fixed with Appwrite 0.15 and later. Feel free to create a new issue if it still persists

**Comment by markosole** [comment]
I was having issues with latest version of Appwrite deployed on Alpine linux instance. I could not make it work and then just simply deploying to Ubuntu problem is gone. Not for those who are in production already and need to migrate :/ 

BTW: If anyone knows solution that works (or suggestion to try) on Alpine, I can give it a go and test further.
